src/Component/DataSource/Driver/Elastica/FieldType/DateTime.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/**
+ * (c) FSi sp. z o.o. <info@fsi.pl>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace FSi\Component\DataSource\Driver\Elastica\FieldType;
+
+use DateTimeInterface;
+use Elastica\Query\BoolQuery;
+use Elastica\Query\Exists;
+use Elastica\Query\Range;
+use FSi\Component\DataSource\Driver\Elastica\Exception\ElasticaDriverException;
+use FSi\Component\DataSource\Field\FieldInterface;
+use FSi\Component\DataSource\Field\Type\DateTimeTypeInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class DateTime extends AbstractField implements DateTimeTypeInterface
+{
+    public function buildQuery(BoolQuery $query, BoolQuery $filter, FieldInterface $field): void
+    {
+        $data = $field->getParameter();
+        if ($this->isEmpty($data)) {
+            return;
+        }
+
+        $fieldPath = $field->getOption('field');
+
+        if ($field->getOption('comparison') === 'eq') {
+            $formattedDate = $data->format($this->getFormat());
+            $filter->addMust(
+                new Range(
+                    $fieldPath,
+                    ['gte' => $formattedDate, 'lte' => $formattedDate]
+                )
+            );
+        } elseif (in_array($field->getOption('comparison'), ['lt', 'lte', 'gt', 'gte'])) {
+            $filter->addMust(
+                new Range(
+                    $fieldPath,
+                    [$field->getOption('comparison') => $data->format($this->getFormat())]
+                )
+            );
+        } elseif ($field->getOption('comparison') === 'between') {
+            if (!is_array($data)) {
```

Comment:
This seems to be a common check, we could make an assertion class for it that would throw an exception when needed.